go/noise/laplace_noise_test.go

```diff
@@ -172,6 +172,525 @@ func TestDeltaForThresholdLaplace(t *testing.T) {
+			lambda: 3,
+			prob:   0.999999971111257,
+			want:   50,
+		},
+	} {
+		got := inverseCDFLaplace(tc.mean, tc.lambda, tc.prob)
+		if !approxEqual(got, tc.want) {
+			t.Errorf("TestInverseCDFLaplace(%f,%f,%f)=%0.12f, want %0.12f, desc: %s", tc.mean, tc.lambda,
+				tc.prob, got, tc.want, tc.desc)
+		}
+	}
+
+}
+
+func TestConfidenceIntervalLaplace(t *testing.T) {
+	// Tests for getConfidenceIntervalLaplace function
+	for _, tc := range []struct {
+		desc                    string
+		noisedValue             float64
+		lambda, confidenceLevel float64
+		want                    ConfidenceInterval
+	}{
+		// 4 Random pre-calculated tests.
+		{
+			desc:            "Random test",
+			noisedValue:     13,
+			lambda:          27.33333333333,
+			confidenceLevel: 0.95,
+			want:            ConfidenceInterval{-68.88334881, 94.88334881},
+		},
+		{
+			desc:            "Random test",
+			noisedValue:     83.1235,
+			lambda:          60,
+			confidenceLevel: 0.76,
+			want:            ConfidenceInterval{-2.503481338, 168.7504813},
+		},
+		{
+			desc:            "Random test",
+			noisedValue:     5,
+			lambda:          6.6666666666667,
+			confidenceLevel: 0.4,
+			want:            ConfidenceInterval{1.594495842, 8.405504158},
+		},
+		{
+			desc:            "Random test",
+			noisedValue:     65.4621,
+			lambda:          700,
+			confidenceLevel: 0.2,
+			want:            ConfidenceInterval{-90.73838592, 221.6625859},
+		},
+		// Near 0 and 1 confidence levels.
+		{
+			desc:            "Low confidence level",
+			noisedValue:     50,
+			lambda:          10,
+			confidenceLevel: 0.01,
+			want:            ConfidenceInterval{49.89949664, 50.1005033595},
+		},
+		{
+			desc:            "High confidence level",
+			noisedValue:     50,
+			lambda:          10,
+			confidenceLevel: 0.99,
+			want:            ConfidenceInterval{3.94829814, 96.05170186},
+		},
+	} {
+		got := getConfidenceIntervalLaplace(tc.noisedValue, tc.lambda, tc.confidenceLevel)
+		if !approxEqual(got.LowerBound, tc.want.LowerBound) {
+			t.Errorf("TestConfidenceIntervalLaplace(%f, %f, %f)=%0.10f, want %0.10f, desc %s, LowerBound is not equal",
+				tc.noisedValue, tc.lambda, tc.confidenceLevel,
+				got.LowerBound, tc.want.LowerBound, tc.desc)
+		}
+		if !approxEqual(got.UpperBound, tc.want.UpperBound) {
+			t.Errorf("TestConfidenceIntervalLaplace(%f, %f, %f)=%0.10f, want %0.10f, desc %s, UpperBound is not equal",
+				tc.noisedValue, tc.lambda, tc.confidenceLevel,
+				got.UpperBound, tc.want.UpperBound, tc.desc)
+		}
+	}
+
+}
+
+func TestConfidenceIntervalFloat64(t *testing.T) {
+	for _, tc := range []struct {
+		desc                                             string
+		noisedValue                                      float64
+		l0Sensitivity                                    int64
+		lInfSensitivity, epsilon, confidenceLevel, delta float64
+		want                                             ConfidenceInterval
+		wantErr                                          bool
+	}{
+		{
```

Comment:
All excpet for one test are argument checks. For the sake of readability and maintainability, I would split this into 2 tests, one for argument checking and one for checking the output (and then maybe add one or two more tests for the second kind of test)

This way, you may also be able to thest the parameters for the int64 and float64 version in the same test, since they should by design have the exact same requirements on their parameters (but ask maria or mirac first if this is good practice)

Reply:
I have already discussed this with them before writing the tests. It was better to split the argument checking. "Each test should be concerned only with one function even if they are similar since they might change later on." This was the argument for this approach.